lib/logging.js

```diff
@@ -6,6 +6,17 @@
 
 'use strict';
 
+/**
+ * @typedef {Object} Logger
+ * @property {function(Record<string, any> | string, ...any): undefined} log
+ * @property {function(Record<string, any> | string, ...any): undefined} info
+ * @property {function(Record<string, any> | string, ...any): undefined} warn
+ * @property {function(Record<string, any> | string, ...any): undefined} error
+ * @property {function(Record<string, any> | string, ...any): undefined} fatal
+ * @property {function(Record<string, any> | string, ...any): undefined} debug
+ * @property {function(Record<string, any> | string, ...any): undefined} trace
```

Comment:
nit: There isn't a `<logger>.log(...)` function.  ([Technically](https://github.com/pinojs/pino/blob/a47c8572ec9f29a08e60e8bb2a735cd1a0342151/pino.d.ts#L211-L214) there is a `.silent(...)` with Pino, but let's not mentioned that one here in our type.)
nit2: My OCD would be happier if these were listed in level order, e.g.: fatal, error, warn, info, debug, trace. :)

Reply:
let's make your OCD happy :)